bindings/wasm/lib/export.ts

```diff
@@ -477,185 +174,147 @@ function cloneNodeNewMaterial(
-  const mesh = doc.createMesh();
-  toNode.setMesh(mesh);
-  fromNode.getMesh()!.listPrimitives().forEach((primitive) => {
+  const oldMesh = fromNode.getMesh()!;
+  const newMesh = doc.createMesh();
+  toNode.setMesh(newMesh);
+  oldMesh.listPrimitives().forEach((primitive) => {
     const newPrimitive = primitive.clone();
     if (primitive.getMaterial() === oldBackupMaterial) {
       newPrimitive.setMaterial(backupMaterial);
     }
-    mesh.addPrimitive(newPrimitive);
+    newMesh.addPrimitive(newPrimitive);
   });
+  // Track cloned meshes for easier export, later.
+  newMesh.setExtras({clonedFrom: oldMesh});
 }
 
 function createNodeFromCache(
-    doc: Document, to3MF: To3MF, nodeDef: GLTFNode,
+    doc: Document, nodeDef: GLTFNode,
     manifold2node: Map<Manifold, Map<GLTFMaterial, Node>>): Node {
   const node = createGLTFnode(doc, nodeDef);
   const {manifold} = nodeDef;
-  if (manifold != null) {
+  if (manifold) {
+    // Animation Morph
     setMorph(doc, node, manifold);
     const backupMaterial = getBackupMaterial(nodeDef);
     const cachedNodes = manifold2node.get(manifold);
+
     if (cachedNodes == null) {
-      addMesh(doc, to3MF, node, manifold, backupMaterial);
+      // Cache miss.
+      addMesh(doc, node, manifold, backupMaterial);
       const cache = new Map<GLTFMaterial, Node>();
       cache.set(backupMaterial, node);
       manifold2node.set(manifold, cache);
+
     } else {
+      // Cache hit...
       const cachedNode = cachedNodes.get(backupMaterial);
       if (cachedNode == null) {
+        // ...but not for this material.
         const [oldBackupMaterial, oldNode] =
             cachedNodes.entries().next().value!;
         cloneNodeNewMaterial(
             doc, node, oldNode, getCachedMaterial(doc, backupMaterial),
             getCachedMaterial(doc, oldBackupMaterial));
         cachedNodes.set(backupMaterial, node);
+
       } else {
+        // ...for this exact material.
         cloneNode(node, cachedNode);
       }
     }
   }
-
-  object2globalID.set(nodeDef, nextGlobalID);
-  to3MF.components.push({
-    id: `${nextGlobalID++}`,
-    name: nodeDef.name,
-    children:
-        manifold == null ? [] : [{objectID: `${object2globalID.get(manifold)}`}]
-  });
   return node;
 }
 
-export async function exportModels(
-    defaults: GlobalDefaults, manifold?: Manifold) {
-  const t0 = performance.now();
-
+function parseOptions(defaults: GlobalDefaults) {
   Object.assign(globalDefaults, GLOBAL_DEFAULTS);
   Object.assign(globalDefaults, defaults);
+}
 
-  const doc = new Document();
+function createWrapper(doc: Document) {
   const halfRoot2 = Math.sqrt(2) / 2;
   const mm2m = 1 / 1000;
   const wrapper = doc.createNode('wrapper')
                       .setRotation([-halfRoot2, 0, 0, halfRoot2])
                       .setScale([mm2m, mm2m, mm2m]);
   doc.createScene().addChild(wrapper);
+  return wrapper
+}
+
+export function manifoldToGLTFDoc(
+    manifold: Manifold, defaults: GlobalDefaults) {
```

Comment:
Could this function just create a GLTFNode from the manifold and then call `GLTFNodesToGLTFDoc` instead? Reduce duplication a bit - not a big deal though.

Reply:
Makes sense.  It's really more of a helper for when users _aren't_ using GLTFNodes from the outset.